src/Extractor/LinkCollectionExtractor.php

```diff
@@ -19,34 +19,25 @@ class LinkCollectionExtractor implements LinkCollectionExtractorInterface
      */
     protected $linkExtractor;
 
-    /**
-     * @param LinkExtractorInterface $linkExtractor
-     */
     public function __construct(LinkExtractorInterface $linkExtractor)
     {
         $this->setLinkExtractor($linkExtractor);
     }
 
-    /**
-     * @return LinkExtractorInterface
-     */
-    public function getLinkExtractor()
+    public function getLinkExtractor(): LinkExtractorInterface
```

Comment:
Please revert all these changes because if the API for this class is changed we have a BC break here. 
And the topic of this pull request is the support of PHP 8, everything else must be done in a separate pull request.

Thanks in advance!